bin/faucet/src/server/pow.rs

```diff
@@ -0,0 +1,107 @@
+use std::{
+    sync::LazyLock,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use axum::{Json, response::IntoResponse};
+use miden_tx::utils::ToHex;
+use rand::{Rng, rng};
+use serde::Serialize;
+use sha3::{Digest, Sha3_256};
+
+const DIFFICULTY: u64 = 5;
+const SERVER_TIMESTAMP_TOLERANCE_SECONDS: u64 = 30;
+
+static SERVER_SALT: LazyLock<String> = LazyLock::new(|| {
+    std::env::var("MIDEN_FAUCET_SERVER_SALT").expect("MIDEN_FAUCET_SERVER_SALT must be set")
```

Comment:
What are our guidelines around configurations like this? Do most things come through a config file or flags (and env vars via flags)? Wondering about this from a consistency perspective.